src/rust/src/decoder/mod.rs

```diff
@@ -0,0 +1,97 @@
+//! CEA 708 decoder
+//!
+//! This module provides a CEA 708 decoder as defined by ANSI/CTA-708-E R-2018
+
+use crate::bindings::*;
+
+const CCX_DTVCC_MAX_PACKET_LENGTH: u8 = 128;
+
+/// Process cc data to generate dtvcc packet
+///
+/// data has following format:
+///
+/// | Name | Size |
+/// | --- | --- |
+/// | cc_valid | 1 byte |
+/// | cc_type | 1 byte |
+/// | data | 2 bytes |
+///
+/// # Safety
+///
+/// dtvcc and data must not be null pointers
+#[no_mangle]
+pub unsafe extern "C" fn dtvcc_process_cc_data(
```

Comment:
Do we need the whole function to be unsafe? We want as many rust safeguards as possible.
I can live with some additional memory copy (since these are just a few bytes) in return of that.

Reply:
Since we are dereferencing a raw pointer, we have to mark it as unsafe. We can change data to not be a raw pointer, but dtvcc might be bigger to copy

Reply:
What you can do here is have a very small function that copies the data (which would have to be unsafe) and then pass it to the large function.

closed caption data is always a few bytes. It's not going to have a performance impact.

Let's prioritize correct code. Otherwise we'd have no reason to use rust in the first place.

Reply:
Ok